src/coreclr/vm/threaddebugblockinginfo.h

```diff
@@ -65,15 +65,35 @@ class ThreadDebugBlockingInfo
 };
 
 #ifndef DACCESS_COMPILE
+
+// This is the equivalent of the managed ThreadBlockingInfo (see ThreadBlockingInfo.cs), which is used for tracking blocking
+// info from the managed side, similarly to DebugBlockingItem
+struct ThreadBlockingInfo
```

Comment:
The current definition appears to be used by debugger APIs to enumerate the blocking items. It seems there would need to be a second object on the stack for maintaining a separate linked list that contains info only about Monitor blocking, as the current APIs to get info from a blocking object would not work with Lock. I was thinking it might be easier to remove the current definitions in the future after debuggers switch to using the managed surface, and maybe deprecate some of the relevant debugger APIs. Thoughts?

Reply:
The types that get exposed publicly should be CorDebugBlockingObject and CorDebugBlockingReason in the ICorDebug interface. The types exposed in the DAC IDacDbiInterface are not public and can be changed as needed. The code to convert from the internal to public types occurs here: https://github.com/dotnet/runtime/blob/d92ac1f892a7f9c00561db6541671f25f3972eed/src/coreclr/debug/di/rsthread.cpp#L2750.

However if we removed the domain field off the internal type we'd have to repopulate it from somewhere else and it looks like we've never done the work to convert ICorDebug to understand that CoreCLR just has a single AppDomain. Its probably not that hard, but I wouldn't feel right asking you to do that refactoring when the ground work isn't there yet. So I'd say lets not worry about this, it was just a potential for some minor deduplication.

In terms of deprecation we've rarely ever done it in the past but it is possible if supporting the old APIs is ongoing hassle and we've allowed some time to pass so that the new API is easily available. I'd think next release at the earliest unless there is some pressing need. We'd need to remove the usage from our tests, remove code from the runtime, and put out notifications of the breaking change. Thanks Kount!